hub/api/tests/test_api.py

```diff
@@ -502,3 +502,17 @@ def test_empty_dataset():
         ds.create_tensor("z")
         ds = Dataset("test")
         assert list(ds.tensors) == ["x", "y", "z"]
+
+
+def test_like(local_ds):
+    local_ds.create_tensor("image", htype="image", sample_compression="png")
+    local_ds.create_tensor("label", htype="class_label", dtype="uint8")
+
+    new_ds_1 = Dataset(local_ds.path + "_test", like=local_ds)
+    new_ds_2 = Dataset(local_ds.path + "_test_1", like=local_ds.path)
+    assert local_ds.meta.as_dict() == new_ds_2.meta.as_dict()
+    assert new_ds_1.tensors.keys() == local_ds.tensors.keys()
+    assert new_ds_2["image"].meta.htype == local_ds["image"].meta.htype
+    assert new_ds_1["label"].meta.dtype == local_ds["label"].meta.dtype
```

Comment:
please check all properties, not just the htype.

Reply:
i recommend writing an `__eq__` function on `TensorMeta` and `DatasetMeta`, and simply doing

```python
assert ds1.meta == ds2.meta

assert len(ds1.tensors) == len(ds2.tensors)

for tensor_name in ds1.tensors.keys():
    assert ds1[tensor_name].meta == ds2[tensor_name].meta
```